src/connection/sqlJob.ts

```diff
@@ -159,6 +159,25 @@ export class OldSQLJob extends SQLJob {
     return connectResult;
   }
 
+  async setCurrentSchema(schema: string): Promise<QueryResult<any>> {
+    if (schema) {
+      const upperSchema = schema.toUpperCase();
+      const result = await this.execute(`set current schema = ?`, {parameters: [upperSchema]});
+      if (result.success) {
+        const config = getInstance().getConfig();
+        const currentLibrary = config.currentLibrary.toUpperCase();
+    
+        if (upperSchema !== currentLibrary) {
+          config.currentLibrary = upperSchema;
+          await getInstance().setConfig(config);
```

Comment:
You actually don't need to update the Code for IBM i config, just the selected job options.

`this.options.libraries[0] = ``;`

The selected job and the connection config have two separate current libraries.